services/treeware/treeware-trees.service.js

```diff
@@ -0,0 +1,63 @@
+'use strict'
+
+const crypto = require('crypto')
+const Joi = require('@hapi/joi')
+const { metric } = require('../text-formatters')
+const { floorCount } = require('../color-formatters')
+const { BaseJsonService } = require('..')
+
+const apiSchema = Joi.object({
+  total: Joi.number().required(),
+}).required()
+
+module.exports = class TreewareTrees extends BaseJsonService {
+  static get category() {
+    return 'other'
+  }
+
+  static get route() {
+    return {
+      base: 'treeware/trees',
+      pattern: ':owner/:packageName',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Treeware (Trees)',
+        namedParams: { owner: 'stoplightio', packageName: 'spectral' },
+        staticPreview: this.render({ count: 250 }),
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'trees' }
+  }
+
+  static render({ count }) {
+    return { message: metric(count), color: floorCount(count, 10, 50, 100) }
+  }
+
+  async fetch({ reference }) {
+    const url = `https://public.offset.earth/users/treeware/trees?ref=${reference}`
```

Comment:
Can we pass ref using `queryParams` instead of templating here please

Reply:
Didn't know about `queryParams` that sounds much better. :+1:

Reply:
Example on how to achieve this here: https://github.com/badges/shields/blob/097bf87e418e0f4242a6d0a8f90c09104d02fadf/services/twitch/twitch.service.js#L64

Reply:
Thank you, I'll add that in shortly.